model/publishing/PublishingService.php

```diff
@@ -108,6 +108,35 @@ public function callEnvironment($action, RequestInterface $request)
         throw new \common_exception_NotFound('No environment found for action "' . $action . '".');
     }
 
+    /**
+     * @param $actionSearch
+     * @return \core_kernel_classes_Resource|mixed|null
+     * @throws \common_exception_NotFound
+     */
+    public function findByAction($actionSearch)
```

Comment:
Could u use this method in CallEnvironment?

Reply:
Done